### PR TITLE
Renamed OSX to macOS and use curl to get EasyInstall script

### PR DESCRIPTION
**Pull request by Bond-009**
Use curl to get the EasyInstall script because the script itself uses curl.

I didn't change the cmd to get the compile.sh script because it didn't work in my testing.

### CODE REVIEW

File: COMPILING.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ COMPILING
 To compile Cuberite from source, you need the following set of software:
 
  * CMake
- * Platform-specific make tool \(windows would be MSVC, linux/osx GNU make, etc.\)
+ * Platform-specific make tool \(windows would be MSVC, linux/macos GNU make, etc.\)
```
Comment (sphinxc0re) [inline COMPILING.md]: It's `macOS` and should be so on every page mentioning this.

File: README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ There are several ways to obtain Cuberite.
 ##### The EasyInstall script
 This script will download the correct binary from the project site.
 
-    sh -c "$(wget -O - https://download.cuberite.org)"
+    curl -sSfL https://download.cuberite.org | sh
```
Comment (sphinxc0re) [inline README.md]: What is the difference between those two?

Comment (Bond-009) [inline README.md]: Use curl because the script itself uses it and not wget

File: COMPILING.md (hunk 1 above)
Comment (Bond-009) [inline COMPILING.md]: That is what I did

Comment (Pokechu22) [inline COMPILING.md]: I think @sphinxc0re is referring to the capitalization: `macOS` instead of `macos`.

Comment (Bond-009) [inline COMPILING.md]: Oh ok, but Linux and Windows where also lower case

File: README.md (hunk 2 above)
Comment (sphinxc0re) [inline README.md]: :+1:

**Comment by bearbin** [comment]
Looks good, merging.

**Pull request merged**